File: VB/GridView.UnboundColumns/Models/Northwind.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Collections

Namespace GridView.UnboundColumns.Models
	Public NotInheritable Class NorthwindDataProvider
		Private Shared db_Renamed As NorthwindDataContext
		Private Sub New()
		End Sub
		Public Shared ReadOnly Property DB() As NorthwindDataContext
			Get
				If db_Renamed Is Nothing Then
					db_Renamed = New NorthwindDataContext()
				End If
				Return db_Renamed
			End Get
		End Property
		Public Shared Function GetInvoices() As IEnumerable
			Return _
				From invoice In DB.Invoices _
				Join customer In DB.Customers On invoice.CustomerID Equals customer.CustomerID _
				Select New With {Key customer.CompanyName, Key invoice.UnitPrice, Key invoice.Quantity}
		End Function
	End Class
End Namespace

File: CS/GridView.UnboundColumns/Models/Northwind.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Collections;

namespace GridView.UnboundColumns.Models {
    public static class NorthwindDataProvider {
        static NorthwindDataContext db;
        public static NorthwindDataContext DB {
            get {
                if(db == null)
                    db = new NorthwindDataContext();
                return db;
            }
        }
        public static IEnumerable GetInvoices() {
            return from invoice in DB.Invoices
                   join customer in DB.Customers on invoice.CustomerID equals customer.CustomerID
                   select new {
                       customer.CompanyName,
                       invoice.UnitPrice,
                       invoice.Quantity
                   };
        }
    }
}